pink/barriers/barrier.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Stéphane Caron
```

Comment:
This file is mostly original, feel free to update the copyright line to ``Copyright 2024 NAME`` where ``NAME`` will be the name of your university/institution if you are enrolled, or your personal name(s) if you work in an individual capacity.

(In the copyright notice from 2022 above, it's my name as an individual as I joined Inria in 2023.)